python/pyspark/sql/udf.py

```diff
@@ -162,7 +168,8 @@ def wrapper(*args):
         wrapper.func = self.func
         wrapper.returnType = self.returnType
         wrapper.evalType = self.evalType
-        wrapper.asNondeterministic = self.asNondeterministic
+        wrapper.deterministic = self.deterministic
+        wrapper.asNondeterministic = lambda: self.asNondeterministic()._wrapped()
```

Comment:
Can we do:

```python
       wrapper.asNondeterministic = functools.wraps(
           self.asNondeterministic)(lambda: self.asNondeterministic()._wrapped())
```

So that it can produce a proper pydoc when we do `help(udf(lambda: 1, "integer").asNondeterministic)` (not `help(udf(lambda: 1, "integer").asNondeterministic())`.

Reply:
good to know the difference

Reply:
I will leave this unchanged. Maybe you can submit a follow-up PR to address it?

Reply:
Definitely. Will give a try within the following week tho ...